packages/components/src/Modal/GenericModal/GenericModal.tsx

```diff
@@ -78,108 +55,94 @@ export const GenericModal = ({
-      className={className}
-    >
-      <FocusLock
-        disabled={focusLockDisabled}
-        returnFocus={true}
-        // Disabling false positive
-        // eslint-disable-next-line jsx-a11y/no-autofocus
-        autoFocus={false}
-      >
-        <div className={styles.backdropLayer} />
-        {/* Disabling these because we don't want this to be keyboard focusable. Users can use Esc to achieve this with a keyboard.
-         */}
-        {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */}
-        <div
-          className={styles.scrollLayer}
-          ref={(scrollLayerRef): void => {
-            setScrollLayer(scrollLayerRef)
+    <AnimatePresence mode="wait">
+      {isOpen && (
+        <ModalContext.Provider
+          value={{
+            labelledByID,
+            describedByID,
           }}
-          onClick={outsideModalClickHandler}
-          data-testid={`${id}-scrollLayer`}
         >
-          <ModalContext.Provider
-            value={{
-              labelledByID,
-              describedByID,
-            }}
+          <FocusOn
+            className={className}
+            onEscapeKey={onEscapeKeyup}
+            scrollLock={true}
+            // False positive, the linter thinks this is the native `autoFocus` prop
+            // eslint-disable-next-line jsx-a11y/no-autofocus
+            autoFocus={false}
           >
+            <motion.div
+              className={styles.backdropLayer}
+              variants={backdropMotion}
+              initial="hidden"
+              animate="visible"
+              exit="exit"
+            />
+            {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */}
             <div
-              role="dialog"
-              className={styles.modalLayer}
-              aria-labelledby={labelledByID}
-              aria-describedby={describedByID}
-              ref={(modalLayerRef): void => setModalLayer(modalLayerRef)}
-              data-testid={id}
+              ref={scrollLayerRef}
+              className={styles.scrollLayer}
+              data-testid={`${id}-scrollLayer`}
+              onClick={handleOutsideClick}
             >
-              {children}
+              <motion.div
+                ref={modalLayerRef}
+                className={styles.modalLayer}
+                variants={modalMotion}
+                initial="hidden"
+                animate="visible"
+                exit="exit"
+                data-testid={id}
+                onAnimationComplete={(latest) => {
+                  if (latest === 'visible') {
+                    handleAfterOpen()
+                    return
+                  }
+                  handleAfterClose()
+                }}
+              >
+                <div
+                  ref={modalRef}
+                  className={classNames(styles.modal, size === 'large' && styles.large)}
```

Comment:
I reckon we move the `className` prop down to here since it won't do anything on the FocusOn container